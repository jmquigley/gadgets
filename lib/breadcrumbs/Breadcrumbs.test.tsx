'use strict';

import {EnumValues as ev} from 'enum-values';
import {mount} from 'enzyme';
import * as React from 'react';
import {
	Breadcrumbs,
	Crumbs,
	getDefaultBreadcrumbsProps,
	Sizing
} from '../../dist/bundle';

const items: Crumbs[] = [
	{name: 'name1', uri: 'http://www.example1.com'},
	{name: 'name2', uri: 'http://www.example2.com'},
	{name: 'name3', uri: 'http://www.example3.com'}
];

test('Test retrieval of Breadcrumbs props object', () => {
	const props = getDefaultBreadcrumbsProps();

	expect(props).toBeDefined();
	expect(props).toMatchSnapshot();
});

for (const sizing of ev.getNames(Sizing)) {
	test(`Create a simple Breadcrumbs control with defaults (${sizing})`, () => {
		const ctl = mount(
			<Breadcrumbs
				items={items}
				sizing={sizing}
			/>
		);

		expect(ctl).toBeDefined();
		expect(ctl).toMatchSnapshot();
	});
}

test('Create a Breadcrumbs component with custom icon and chevron', () => {
	const ctl = mount(
		<Breadcrumbs
			chevron="arrow-right"
			icon="pied-piper"
			items={items}
		/>
	);

	expect(ctl).toBeDefined();
	expect(ctl).toMatchSnapshot();
});

test('Test making the Breadcrumbs invisible', () => {
	const ctl = mount(
		<Breadcrumbs
			items={items}
			visible={false}
		/>
	);

	expect(ctl).toBeDefined();
	expect(ctl).toMatchSnapshot();
});

test('Test disabling the Breadcrumbs control', () => {
	const select = jest.fn();
	const ctl = mount(
		<Breadcrumbs
			disabled
			items={items}
			onSelect={select}
		/>
	);

	expect(ctl).toBeDefined();
	expect(ctl).toMatchSnapshot();
	expect(select).not.toHaveBeenCalled();
});

test('Test Breadcrumbs select event', () => {
	const select = jest.fn();
	const ctl = mount(
		<Breadcrumbs
			items={items}
			onSelect={select}
		/>
	);

	expect(ctl).toBeDefined();
	expect(ctl).toMatchSnapshot();

	ctl.find('.ui-breadcrumbs-name').first().simulate('click');
	expect(select).toHaveBeenCalled();
	expect(select).toHaveBeenCalledWith('name1', 'http://www.example1.com');
});
