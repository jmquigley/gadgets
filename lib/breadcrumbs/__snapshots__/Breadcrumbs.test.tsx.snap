// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create a Breadcrumbs component with custom icon and chevron 1`] = `
<Breadcrumbs
  backgroundColor={null}
  border={null}
  borderColor={null}
  bottom={null}
  chevron="arrow-right"
  className={null}
  color={null}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  hidden={false}
  icon="pied-piper"
  id={null}
  items={
    Array [
      Object {
        "name": "name1",
        "uri": "http://www.example1.com",
      },
      Object {
        "name": "name2",
        "uri": "http://www.example2.com",
      },
      Object {
        "name": "name3",
        "uri": "http://www.example3.com",
      },
    ]
  }
  left={null}
  location="none"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noborder={false}
  noedit={false}
  nohover={false}
  nopropagation={false}
  notheme={false}
  notooltip={false}
  obj="Breadcrumbs"
  onSelect={[Function]}
  padding={null}
  ripple={false}
  selected={false}
  sizing="normal"
  style={Object {}}
  testing={true}
  theme={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#c0c0c0",
      "buttonColor": "#666666",
      "chevronColor": "#cccccc",
      "color": "#000000",
      "fontFamily": "OpenSans, Georgia, serif",
      "headerBackgroundColor": "#004358",
      "headerForegroundColor": "#ffffff",
      "headerHoverColor": "#ffe11a",
      "hoverColor": "#c0c0c0",
      "inputBorderColor": "#c0c0c0",
      "itemHoverColor": "#bedb39",
      "outlineColor": "#51cbee",
      "searchFocus": "#51cbee",
      "searchMatch": "#fd7400",
      "selectedBackgroundColor": "#1f8a70",
      "selectedForegroundColor": "#ffe11a",
      "titleBarBackgroundColor": "#004358",
      "titleBarForegroundColor": "#ffe11a",
      "tooltipBackgroundColor": "#708090",
      "tooltipForegroundColor": "#ffffff",
      "transitionDelay": "0.33s",
    }
  }
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
/>
`;

exports[`Create a simple Breadcrumbs control with defaults (inherit) 1`] = `
<Breadcrumbs
  backgroundColor={null}
  border={null}
  borderColor={null}
  bottom={null}
  chevron="chevron-right"
  className={null}
  color={null}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  hidden={false}
  icon="paperclip"
  id={null}
  items={
    Array [
      Object {
        "name": "name1",
        "uri": "http://www.example1.com",
      },
      Object {
        "name": "name2",
        "uri": "http://www.example2.com",
      },
      Object {
        "name": "name3",
        "uri": "http://www.example3.com",
      },
    ]
  }
  left={null}
  location="none"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noborder={false}
  noedit={false}
  nohover={false}
  nopropagation={false}
  notheme={false}
  notooltip={false}
  obj="Breadcrumbs"
  onSelect={[Function]}
  padding={null}
  ripple={false}
  selected={false}
  sizing="inherit"
  style={Object {}}
  testing={true}
  theme={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#c0c0c0",
      "buttonColor": "#666666",
      "chevronColor": "#cccccc",
      "color": "#000000",
      "fontFamily": "OpenSans, Georgia, serif",
      "headerBackgroundColor": "#004358",
      "headerForegroundColor": "#ffffff",
      "headerHoverColor": "#ffe11a",
      "hoverColor": "#c0c0c0",
      "inputBorderColor": "#c0c0c0",
      "itemHoverColor": "#bedb39",
      "outlineColor": "#51cbee",
      "searchFocus": "#51cbee",
      "searchMatch": "#fd7400",
      "selectedBackgroundColor": "#1f8a70",
      "selectedForegroundColor": "#ffe11a",
      "titleBarBackgroundColor": "#004358",
      "titleBarForegroundColor": "#ffe11a",
      "tooltipBackgroundColor": "#708090",
      "tooltipForegroundColor": "#ffffff",
      "transitionDelay": "0.33s",
    }
  }
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
/>
`;

exports[`Create a simple Breadcrumbs control with defaults (large) 1`] = `
<Breadcrumbs
  backgroundColor={null}
  border={null}
  borderColor={null}
  bottom={null}
  chevron="chevron-right"
  className={null}
  color={null}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  hidden={false}
  icon="paperclip"
  id={null}
  items={
    Array [
      Object {
        "name": "name1",
        "uri": "http://www.example1.com",
      },
      Object {
        "name": "name2",
        "uri": "http://www.example2.com",
      },
      Object {
        "name": "name3",
        "uri": "http://www.example3.com",
      },
    ]
  }
  left={null}
  location="none"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noborder={false}
  noedit={false}
  nohover={false}
  nopropagation={false}
  notheme={false}
  notooltip={false}
  obj="Breadcrumbs"
  onSelect={[Function]}
  padding={null}
  ripple={false}
  selected={false}
  sizing="large"
  style={Object {}}
  testing={true}
  theme={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#c0c0c0",
      "buttonColor": "#666666",
      "chevronColor": "#cccccc",
      "color": "#000000",
      "fontFamily": "OpenSans, Georgia, serif",
      "headerBackgroundColor": "#004358",
      "headerForegroundColor": "#ffffff",
      "headerHoverColor": "#ffe11a",
      "hoverColor": "#c0c0c0",
      "inputBorderColor": "#c0c0c0",
      "itemHoverColor": "#bedb39",
      "outlineColor": "#51cbee",
      "searchFocus": "#51cbee",
      "searchMatch": "#fd7400",
      "selectedBackgroundColor": "#1f8a70",
      "selectedForegroundColor": "#ffe11a",
      "titleBarBackgroundColor": "#004358",
      "titleBarForegroundColor": "#ffe11a",
      "tooltipBackgroundColor": "#708090",
      "tooltipForegroundColor": "#ffffff",
      "transitionDelay": "0.33s",
    }
  }
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
/>
`;

exports[`Create a simple Breadcrumbs control with defaults (medium) 1`] = `
<Breadcrumbs
  backgroundColor={null}
  border={null}
  borderColor={null}
  bottom={null}
  chevron="chevron-right"
  className={null}
  color={null}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  hidden={false}
  icon="paperclip"
  id={null}
  items={
    Array [
      Object {
        "name": "name1",
        "uri": "http://www.example1.com",
      },
      Object {
        "name": "name2",
        "uri": "http://www.example2.com",
      },
      Object {
        "name": "name3",
        "uri": "http://www.example3.com",
      },
    ]
  }
  left={null}
  location="none"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noborder={false}
  noedit={false}
  nohover={false}
  nopropagation={false}
  notheme={false}
  notooltip={false}
  obj="Breadcrumbs"
  onSelect={[Function]}
  padding={null}
  ripple={false}
  selected={false}
  sizing="medium"
  style={Object {}}
  testing={true}
  theme={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#c0c0c0",
      "buttonColor": "#666666",
      "chevronColor": "#cccccc",
      "color": "#000000",
      "fontFamily": "OpenSans, Georgia, serif",
      "headerBackgroundColor": "#004358",
      "headerForegroundColor": "#ffffff",
      "headerHoverColor": "#ffe11a",
      "hoverColor": "#c0c0c0",
      "inputBorderColor": "#c0c0c0",
      "itemHoverColor": "#bedb39",
      "outlineColor": "#51cbee",
      "searchFocus": "#51cbee",
      "searchMatch": "#fd7400",
      "selectedBackgroundColor": "#1f8a70",
      "selectedForegroundColor": "#ffe11a",
      "titleBarBackgroundColor": "#004358",
      "titleBarForegroundColor": "#ffe11a",
      "tooltipBackgroundColor": "#708090",
      "tooltipForegroundColor": "#ffffff",
      "transitionDelay": "0.33s",
    }
  }
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
/>
`;

exports[`Create a simple Breadcrumbs control with defaults (normal) 1`] = `
<Breadcrumbs
  backgroundColor={null}
  border={null}
  borderColor={null}
  bottom={null}
  chevron="chevron-right"
  className={null}
  color={null}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  hidden={false}
  icon="paperclip"
  id={null}
  items={
    Array [
      Object {
        "name": "name1",
        "uri": "http://www.example1.com",
      },
      Object {
        "name": "name2",
        "uri": "http://www.example2.com",
      },
      Object {
        "name": "name3",
        "uri": "http://www.example3.com",
      },
    ]
  }
  left={null}
  location="none"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noborder={false}
  noedit={false}
  nohover={false}
  nopropagation={false}
  notheme={false}
  notooltip={false}
  obj="Breadcrumbs"
  onSelect={[Function]}
  padding={null}
  ripple={false}
  selected={false}
  sizing="normal"
  style={Object {}}
  testing={true}
  theme={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#c0c0c0",
      "buttonColor": "#666666",
      "chevronColor": "#cccccc",
      "color": "#000000",
      "fontFamily": "OpenSans, Georgia, serif",
      "headerBackgroundColor": "#004358",
      "headerForegroundColor": "#ffffff",
      "headerHoverColor": "#ffe11a",
      "hoverColor": "#c0c0c0",
      "inputBorderColor": "#c0c0c0",
      "itemHoverColor": "#bedb39",
      "outlineColor": "#51cbee",
      "searchFocus": "#51cbee",
      "searchMatch": "#fd7400",
      "selectedBackgroundColor": "#1f8a70",
      "selectedForegroundColor": "#ffe11a",
      "titleBarBackgroundColor": "#004358",
      "titleBarForegroundColor": "#ffe11a",
      "tooltipBackgroundColor": "#708090",
      "tooltipForegroundColor": "#ffffff",
      "transitionDelay": "0.33s",
    }
  }
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
/>
`;

exports[`Create a simple Breadcrumbs control with defaults (small) 1`] = `
<Breadcrumbs
  backgroundColor={null}
  border={null}
  borderColor={null}
  bottom={null}
  chevron="chevron-right"
  className={null}
  color={null}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  hidden={false}
  icon="paperclip"
  id={null}
  items={
    Array [
      Object {
        "name": "name1",
        "uri": "http://www.example1.com",
      },
      Object {
        "name": "name2",
        "uri": "http://www.example2.com",
      },
      Object {
        "name": "name3",
        "uri": "http://www.example3.com",
      },
    ]
  }
  left={null}
  location="none"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noborder={false}
  noedit={false}
  nohover={false}
  nopropagation={false}
  notheme={false}
  notooltip={false}
  obj="Breadcrumbs"
  onSelect={[Function]}
  padding={null}
  ripple={false}
  selected={false}
  sizing="small"
  style={Object {}}
  testing={true}
  theme={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#c0c0c0",
      "buttonColor": "#666666",
      "chevronColor": "#cccccc",
      "color": "#000000",
      "fontFamily": "OpenSans, Georgia, serif",
      "headerBackgroundColor": "#004358",
      "headerForegroundColor": "#ffffff",
      "headerHoverColor": "#ffe11a",
      "hoverColor": "#c0c0c0",
      "inputBorderColor": "#c0c0c0",
      "itemHoverColor": "#bedb39",
      "outlineColor": "#51cbee",
      "searchFocus": "#51cbee",
      "searchMatch": "#fd7400",
      "selectedBackgroundColor": "#1f8a70",
      "selectedForegroundColor": "#ffe11a",
      "titleBarBackgroundColor": "#004358",
      "titleBarForegroundColor": "#ffe11a",
      "tooltipBackgroundColor": "#708090",
      "tooltipForegroundColor": "#ffffff",
      "transitionDelay": "0.33s",
    }
  }
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
/>
`;

exports[`Create a simple Breadcrumbs control with defaults (xlarge) 1`] = `
<Breadcrumbs
  backgroundColor={null}
  border={null}
  borderColor={null}
  bottom={null}
  chevron="chevron-right"
  className={null}
  color={null}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  hidden={false}
  icon="paperclip"
  id={null}
  items={
    Array [
      Object {
        "name": "name1",
        "uri": "http://www.example1.com",
      },
      Object {
        "name": "name2",
        "uri": "http://www.example2.com",
      },
      Object {
        "name": "name3",
        "uri": "http://www.example3.com",
      },
    ]
  }
  left={null}
  location="none"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noborder={false}
  noedit={false}
  nohover={false}
  nopropagation={false}
  notheme={false}
  notooltip={false}
  obj="Breadcrumbs"
  onSelect={[Function]}
  padding={null}
  ripple={false}
  selected={false}
  sizing="xlarge"
  style={Object {}}
  testing={true}
  theme={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#c0c0c0",
      "buttonColor": "#666666",
      "chevronColor": "#cccccc",
      "color": "#000000",
      "fontFamily": "OpenSans, Georgia, serif",
      "headerBackgroundColor": "#004358",
      "headerForegroundColor": "#ffffff",
      "headerHoverColor": "#ffe11a",
      "hoverColor": "#c0c0c0",
      "inputBorderColor": "#c0c0c0",
      "itemHoverColor": "#bedb39",
      "outlineColor": "#51cbee",
      "searchFocus": "#51cbee",
      "searchMatch": "#fd7400",
      "selectedBackgroundColor": "#1f8a70",
      "selectedForegroundColor": "#ffe11a",
      "titleBarBackgroundColor": "#004358",
      "titleBarForegroundColor": "#ffe11a",
      "tooltipBackgroundColor": "#708090",
      "tooltipForegroundColor": "#ffffff",
      "transitionDelay": "0.33s",
    }
  }
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
/>
`;

exports[`Create a simple Breadcrumbs control with defaults (xsmall) 1`] = `
<Breadcrumbs
  backgroundColor={null}
  border={null}
  borderColor={null}
  bottom={null}
  chevron="chevron-right"
  className={null}
  color={null}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  hidden={false}
  icon="paperclip"
  id={null}
  items={
    Array [
      Object {
        "name": "name1",
        "uri": "http://www.example1.com",
      },
      Object {
        "name": "name2",
        "uri": "http://www.example2.com",
      },
      Object {
        "name": "name3",
        "uri": "http://www.example3.com",
      },
    ]
  }
  left={null}
  location="none"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noborder={false}
  noedit={false}
  nohover={false}
  nopropagation={false}
  notheme={false}
  notooltip={false}
  obj="Breadcrumbs"
  onSelect={[Function]}
  padding={null}
  ripple={false}
  selected={false}
  sizing="xsmall"
  style={Object {}}
  testing={true}
  theme={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#c0c0c0",
      "buttonColor": "#666666",
      "chevronColor": "#cccccc",
      "color": "#000000",
      "fontFamily": "OpenSans, Georgia, serif",
      "headerBackgroundColor": "#004358",
      "headerForegroundColor": "#ffffff",
      "headerHoverColor": "#ffe11a",
      "hoverColor": "#c0c0c0",
      "inputBorderColor": "#c0c0c0",
      "itemHoverColor": "#bedb39",
      "outlineColor": "#51cbee",
      "searchFocus": "#51cbee",
      "searchMatch": "#fd7400",
      "selectedBackgroundColor": "#1f8a70",
      "selectedForegroundColor": "#ffe11a",
      "titleBarBackgroundColor": "#004358",
      "titleBarForegroundColor": "#ffe11a",
      "tooltipBackgroundColor": "#708090",
      "tooltipForegroundColor": "#ffffff",
      "transitionDelay": "0.33s",
    }
  }
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
/>
`;

exports[`Create a simple Breadcrumbs control with defaults (xxlarge) 1`] = `
<Breadcrumbs
  backgroundColor={null}
  border={null}
  borderColor={null}
  bottom={null}
  chevron="chevron-right"
  className={null}
  color={null}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  hidden={false}
  icon="paperclip"
  id={null}
  items={
    Array [
      Object {
        "name": "name1",
        "uri": "http://www.example1.com",
      },
      Object {
        "name": "name2",
        "uri": "http://www.example2.com",
      },
      Object {
        "name": "name3",
        "uri": "http://www.example3.com",
      },
    ]
  }
  left={null}
  location="none"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noborder={false}
  noedit={false}
  nohover={false}
  nopropagation={false}
  notheme={false}
  notooltip={false}
  obj="Breadcrumbs"
  onSelect={[Function]}
  padding={null}
  ripple={false}
  selected={false}
  sizing="xxlarge"
  style={Object {}}
  testing={true}
  theme={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#c0c0c0",
      "buttonColor": "#666666",
      "chevronColor": "#cccccc",
      "color": "#000000",
      "fontFamily": "OpenSans, Georgia, serif",
      "headerBackgroundColor": "#004358",
      "headerForegroundColor": "#ffffff",
      "headerHoverColor": "#ffe11a",
      "hoverColor": "#c0c0c0",
      "inputBorderColor": "#c0c0c0",
      "itemHoverColor": "#bedb39",
      "outlineColor": "#51cbee",
      "searchFocus": "#51cbee",
      "searchMatch": "#fd7400",
      "selectedBackgroundColor": "#1f8a70",
      "selectedForegroundColor": "#ffe11a",
      "titleBarBackgroundColor": "#004358",
      "titleBarForegroundColor": "#ffe11a",
      "tooltipBackgroundColor": "#708090",
      "tooltipForegroundColor": "#ffffff",
      "transitionDelay": "0.33s",
    }
  }
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
/>
`;

exports[`Create a simple Breadcrumbs control with defaults (xxsmall) 1`] = `
<Breadcrumbs
  backgroundColor={null}
  border={null}
  borderColor={null}
  bottom={null}
  chevron="chevron-right"
  className={null}
  color={null}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  hidden={false}
  icon="paperclip"
  id={null}
  items={
    Array [
      Object {
        "name": "name1",
        "uri": "http://www.example1.com",
      },
      Object {
        "name": "name2",
        "uri": "http://www.example2.com",
      },
      Object {
        "name": "name3",
        "uri": "http://www.example3.com",
      },
    ]
  }
  left={null}
  location="none"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noborder={false}
  noedit={false}
  nohover={false}
  nopropagation={false}
  notheme={false}
  notooltip={false}
  obj="Breadcrumbs"
  onSelect={[Function]}
  padding={null}
  ripple={false}
  selected={false}
  sizing="xxsmall"
  style={Object {}}
  testing={true}
  theme={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#c0c0c0",
      "buttonColor": "#666666",
      "chevronColor": "#cccccc",
      "color": "#000000",
      "fontFamily": "OpenSans, Georgia, serif",
      "headerBackgroundColor": "#004358",
      "headerForegroundColor": "#ffffff",
      "headerHoverColor": "#ffe11a",
      "hoverColor": "#c0c0c0",
      "inputBorderColor": "#c0c0c0",
      "itemHoverColor": "#bedb39",
      "outlineColor": "#51cbee",
      "searchFocus": "#51cbee",
      "searchMatch": "#fd7400",
      "selectedBackgroundColor": "#1f8a70",
      "selectedForegroundColor": "#ffe11a",
      "titleBarBackgroundColor": "#004358",
      "titleBarForegroundColor": "#ffe11a",
      "tooltipBackgroundColor": "#708090",
      "tooltipForegroundColor": "#ffffff",
      "transitionDelay": "0.33s",
    }
  }
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
/>
`;

exports[`Test Breadcrumbs select event 1`] = `
<Breadcrumbs
  backgroundColor={null}
  border={null}
  borderColor={null}
  bottom={null}
  chevron="chevron-right"
  className={null}
  color={null}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  hidden={false}
  icon="paperclip"
  id={null}
  items={
    Array [
      Object {
        "name": "name1",
        "uri": "http://www.example1.com",
      },
      Object {
        "name": "name2",
        "uri": "http://www.example2.com",
      },
      Object {
        "name": "name3",
        "uri": "http://www.example3.com",
      },
    ]
  }
  left={null}
  location="none"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noborder={false}
  noedit={false}
  nohover={false}
  nopropagation={false}
  notheme={false}
  notooltip={false}
  obj="Breadcrumbs"
  onSelect={[MockFunction]}
  padding={null}
  ripple={false}
  selected={false}
  sizing="normal"
  style={Object {}}
  testing={true}
  theme={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#c0c0c0",
      "buttonColor": "#666666",
      "chevronColor": "#cccccc",
      "color": "#000000",
      "fontFamily": "OpenSans, Georgia, serif",
      "headerBackgroundColor": "#004358",
      "headerForegroundColor": "#ffffff",
      "headerHoverColor": "#ffe11a",
      "hoverColor": "#c0c0c0",
      "inputBorderColor": "#c0c0c0",
      "itemHoverColor": "#bedb39",
      "outlineColor": "#51cbee",
      "searchFocus": "#51cbee",
      "searchMatch": "#fd7400",
      "selectedBackgroundColor": "#1f8a70",
      "selectedForegroundColor": "#ffe11a",
      "titleBarBackgroundColor": "#004358",
      "titleBarForegroundColor": "#ffe11a",
      "tooltipBackgroundColor": "#708090",
      "tooltipForegroundColor": "#ffffff",
      "transitionDelay": "0.33s",
    }
  }
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
/>
`;

exports[`Test disabling the Breadcrumbs control 1`] = `
<Breadcrumbs
  backgroundColor={null}
  border={null}
  borderColor={null}
  bottom={null}
  chevron="chevron-right"
  className={null}
  color={null}
  controlled={true}
  direction="right"
  disabled={true}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  hidden={false}
  icon="paperclip"
  id={null}
  items={
    Array [
      Object {
        "name": "name1",
        "uri": "http://www.example1.com",
      },
      Object {
        "name": "name2",
        "uri": "http://www.example2.com",
      },
      Object {
        "name": "name3",
        "uri": "http://www.example3.com",
      },
    ]
  }
  left={null}
  location="none"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noborder={false}
  noedit={false}
  nohover={false}
  nopropagation={false}
  notheme={false}
  notooltip={false}
  obj="Breadcrumbs"
  onSelect={[MockFunction]}
  padding={null}
  ripple={false}
  selected={false}
  sizing="normal"
  style={Object {}}
  testing={true}
  theme={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#c0c0c0",
      "buttonColor": "#666666",
      "chevronColor": "#cccccc",
      "color": "#000000",
      "fontFamily": "OpenSans, Georgia, serif",
      "headerBackgroundColor": "#004358",
      "headerForegroundColor": "#ffffff",
      "headerHoverColor": "#ffe11a",
      "hoverColor": "#c0c0c0",
      "inputBorderColor": "#c0c0c0",
      "itemHoverColor": "#bedb39",
      "outlineColor": "#51cbee",
      "searchFocus": "#51cbee",
      "searchMatch": "#fd7400",
      "selectedBackgroundColor": "#1f8a70",
      "selectedForegroundColor": "#ffe11a",
      "titleBarBackgroundColor": "#004358",
      "titleBarForegroundColor": "#ffe11a",
      "tooltipBackgroundColor": "#708090",
      "tooltipForegroundColor": "#ffffff",
      "transitionDelay": "0.33s",
    }
  }
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
/>
`;

exports[`Test making the Breadcrumbs invisible 1`] = `
<Breadcrumbs
  backgroundColor={null}
  border={null}
  borderColor={null}
  bottom={null}
  chevron="chevron-right"
  className={null}
  color={null}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  hidden={false}
  icon="paperclip"
  id={null}
  items={
    Array [
      Object {
        "name": "name1",
        "uri": "http://www.example1.com",
      },
      Object {
        "name": "name2",
        "uri": "http://www.example2.com",
      },
      Object {
        "name": "name3",
        "uri": "http://www.example3.com",
      },
    ]
  }
  left={null}
  location="none"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noborder={false}
  noedit={false}
  nohover={false}
  nopropagation={false}
  notheme={false}
  notooltip={false}
  obj="Breadcrumbs"
  onSelect={[Function]}
  padding={null}
  ripple={false}
  selected={false}
  sizing="normal"
  style={Object {}}
  testing={true}
  theme={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#c0c0c0",
      "buttonColor": "#666666",
      "chevronColor": "#cccccc",
      "color": "#000000",
      "fontFamily": "OpenSans, Georgia, serif",
      "headerBackgroundColor": "#004358",
      "headerForegroundColor": "#ffffff",
      "headerHoverColor": "#ffe11a",
      "hoverColor": "#c0c0c0",
      "inputBorderColor": "#c0c0c0",
      "itemHoverColor": "#bedb39",
      "outlineColor": "#51cbee",
      "searchFocus": "#51cbee",
      "searchMatch": "#fd7400",
      "selectedBackgroundColor": "#1f8a70",
      "selectedForegroundColor": "#ffe11a",
      "titleBarBackgroundColor": "#004358",
      "titleBarForegroundColor": "#ffe11a",
      "tooltipBackgroundColor": "#708090",
      "tooltipForegroundColor": "#ffffff",
      "transitionDelay": "0.33s",
    }
  }
  tooltip={null}
  top={null}
  visible={false}
  width={null}
  xcss={null}
/>
`;

exports[`Test retrieval of Breadcrumbs props object 1`] = `
Object {
  "backgroundColor": null,
  "border": null,
  "borderColor": null,
  "bottom": null,
  "chevron": "chevron-right",
  "children": null,
  "className": null,
  "color": null,
  "controlled": true,
  "direction": "right",
  "disabled": false,
  "err": null,
  "errorMessage": null,
  "focus": false,
  "height": null,
  "hidden": false,
  "icon": "paperclip",
  "id": null,
  "items": Array [],
  "left": null,
  "location": "none",
  "maxHeight": null,
  "maxWidth": null,
  "minHeight": null,
  "minWidth": null,
  "noborder": false,
  "noedit": false,
  "nohover": false,
  "nopropagation": false,
  "notheme": false,
  "notooltip": false,
  "obj": "Breadcrumbs",
  "onSelect": [Function],
  "padding": null,
  "ripple": false,
  "selected": false,
  "sizing": "normal",
  "style": Object {},
  "testing": true,
  "theme": Object {
    "backgroundColor": "#ffffff",
    "borderColor": "#c0c0c0",
    "buttonColor": "#666666",
    "chevronColor": "#cccccc",
    "color": "#000000",
    "fontFamily": "OpenSans, Georgia, serif",
    "headerBackgroundColor": "#004358",
    "headerForegroundColor": "#ffffff",
    "headerHoverColor": "#ffe11a",
    "hoverColor": "#c0c0c0",
    "inputBorderColor": "#c0c0c0",
    "itemHoverColor": "#bedb39",
    "outlineColor": "#51cbee",
    "searchFocus": "#51cbee",
    "searchMatch": "#fd7400",
    "selectedBackgroundColor": "#1f8a70",
    "selectedForegroundColor": "#ffe11a",
    "titleBarBackgroundColor": "#004358",
    "titleBarForegroundColor": "#ffe11a",
    "tooltipBackgroundColor": "#708090",
    "tooltipForegroundColor": "#ffffff",
    "transitionDelay": "0.33s",
  },
  "tooltip": null,
  "top": null,
  "visible": true,
  "width": null,
  "xcss": null,
}
`;
